### PR TITLE
MakeMKV plugin for VLC

### PR DESCRIPTION
**Pull request by MatMaul**
Provides MakeMKV features for direct Blu-ray playback in VLC.

### CODE REVIEW

**Comment by nedrichards** [comment]
bot, build org.videolan.VLC.Plugin.makemkv

**Comment by MatMaul** [comment]
runtime-version added and full runtime qualifier removed like in fdkaac.

File: org.videolan.VLC.Plugin.makemkv.json [hunk 1]
```diff
@@ -0,0 +1,90 @@
+{
+  "id": "org.videolan.VLC.Plugin.makemkv",
+  "build-extension": true,
+  "runtime": "org.videolan.VLC",
+  "runtime-version": "stable",
+  "branch": "3-1.6",
+  "sdk": "org.freedesktop.Sdk//1.6",
```
Comment (bilelmoussaoui) [inline org.videolan.VLC.Plugin.makemkv.json]: 1.6? you should use 18.08 instead here

Comment (MatMaul) [reply org.videolan.VLC.Plugin.makemkv.json]: I don't think so, org.videolan.VLC is still on 1.6, via Qt 5.11 Sdk.

**Comment by MatMaul** [comment]
I have no idea what is happening... It fails right away on runtime/sdk checking, and this part is exactly the same as in the fdkaac plugin. Am I missing something stupid ?

File: org.videolan.VLC.Plugin.makemkv.json [hunk 2]
```diff
@@ -0,0 +1,90 @@
+{
+  "id": "org.videolan.VLC.Plugin.makemkv",
+  "build-extension": true,
+  "runtime": "org.videolan.VLC",
+  "runtime-version": "stable",
+  "branch": "3-1.6",
```
Comment (barthalion) [inline org.videolan.VLC.Plugin.makemkv.json]: VLC isn't branched so your plugin shouldn't be either IMHO.

File: org.videolan.VLC.Plugin.makemkv.json [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+  "sdk": "org.freedesktop.Sdk//1.6",
+  "platform-extensions": [
```
Comment (barthalion) [inline org.videolan.VLC.Plugin.makemkv.json]: `sdk-extensions`

**Comment by barthalion** [comment]
Regardless of my comments, I think we are hitting a bug in flatpak-builder. With fixes applied:

```
FB: Running: git config --get user.email
FB: Running: git config --get user.name
Dependency Sdk: org.freedesktop.Sdk//1.6 stable
FB: Running: flatpak --user info org.freedesktop.Sdk//1.6/x86_64/stable --show-commit
FB: Running: flatpak --user update --subpath= -y org.freedesktop.Sdk//1.6/x86_64/stable
Looking for updates...
Dependency Runtime: org.videolan.VLC stable
FB: Running: flatpak --user info org.videolan.VLC/x86_64/stable --show-commit
FB: Running: flatpak --user update --subpath= -y org.videolan.VLC/x86_64/stable
Looking for updates...
Installing in user:
org.videolan.VLC.Locale/x86_64/stable flathub 2aa02fa45af6
Installing for user: org.videolan.VLC.Locale/x86_64/stable from flathub
Warning: Failed to install org.videolan.VLC.Locale/x86_64/stable: Server returned status 503: Service Unavailable
FB: Running: flatpak --user info runtime/org.freedesktop.Sdk//1.6/x86_64/stable --show-metadata
Dependency Extension: org.freedesktop.Sdk.Extension.openjdk10 stable
FB: Running: flatpak --user info org.freedesktop.Sdk.Extension.openjdk10/x86_64/stable --show-commit
FB: Running: flatpak --user install flathub -y org.freedesktop.Sdk.Extension.openjdk10/x86_64/stable
error: Nothing matches org.freedesktop.Sdk.Extension.openjdk10 in remote flathub
Error running (null): Child process exited with code 1
```

flatpak-builder looks for `stable` version of given SDK extension that is set as `branch` first, ignoring `extension//version` notation.

File: org.videolan.VLC.Plugin.makemkv.json (hunk 2 above)
Comment (MatMaul) [reply org.videolan.VLC.Plugin.makemkv.json]: @barthalion Sorry what do you mean by this ? The extension point in VLC is versionned so I need to target 3-18.08 (nevermind the 3-1.6 here, it's 3-18.08 in the current commit).

Comment (MatMaul) [reply org.videolan.VLC.Plugin.makemkv.json]: Or you are talking about `runtime-version` ?
Without it on my computer it defaults to master and fails.

**Comment by MatMaul** [comment]
Flatpak has been fixed for `sdk-extensions` and the JVM dependency moved to #849 anyway.
I just check and it still works on my computer, if someone could trigger a build.
Thanks !

**Comment by nedrichards** [comment]
bot, build org.videolan.VLC.Plugin.makemkv
testbot, build org.videolan.VLC.Plugin.makemkv

**Comment by nedrichards** [comment]
Repository has been created: https://github.com/flathub/org.videolan.VLC.Plugin.makemkv

**Pull request closed without merging**